crates/router/src/core/webhooks/outgoing.rs

```diff
@@ -457,15 +473,40 @@ fn raise_webhooks_analytics_event(
         .and_then(api::OutgoingWebhookContent::get_outgoing_webhook_event_content)
         .or_else(|| get_outgoing_webhook_event_content_from_event_metadata(event.metadata));
 
+    // Fetch updated_event from db
+    let debug = format!("event not found for id: {}", &event_id);
+    event = state
+        .store
+        .find_event_by_merchant_id_event_id(
+            key_manager_state,
+            &merchant_id,
+            &event_id,
+            merchant_key_store,
+        )
+        .await
+        .change_context(errors::StorageError::ValueNotFound(debug.clone()))
+        .attach_printable(debug)?;
+
+    // Get status_code from webhook response
+    let webhook_response: Option<OutgoingWebhookResponseContent> = event
+        .response
+        .map(|res| res.peek().to_owned())
+        .and_then(|res| res.parse_struct("OutgoingWebhookResponseContent").ok());
+    let status_code = webhook_response.and_then(|res| res.status_code);
```

Comment:
1. Let's not propagate the error here with `?`, but instead log it and populate `None` for the `status_code` in the analytics event if we're unable to retrieve the event from the database. We'd like to raise an analytics event, even if we're not able to obtain the status code that the merchant server returned.
2. Use `attach_printable_lazy()` instead of `attach_printable()` and move the `format!()` call within `attach_printable_lazy()`, since `format!()` involves a heap allocation.
3. Log the error if deserializing the response as `OutgoingWebhookResponseContent` fails.
4. Once these changes have been incorporated, the function no longer would be required to return a `Result`.